src/mlir/design.md

```diff
@@ -0,0 +1,158 @@
+# Verona MLIR Generator Design Choices
+
+The MLIR generator sits in between the AST and LLVM IR, and is meant to ease
```

Comment:
Please don't line break in the middle of sentences, it makes reviewing diffs difficult.